cyral/internal/policyset/v1/model.go

```diff
@@ -0,0 +1,147 @@
+package policysetv1
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/cyralinc/terraform-provider-cyral/cyral/utils"
+)
+
+// ChangeInfo represents information about changes to the policy set
+type ChangeInfo struct {
+	Actor     string `json:"actor,omitempty"`
+	ActorType string `json:"actorType,omitempty"`
+	Timestamp string `json:"timestamp,omitempty"`
+}
+
+// ToMap converts ChangeInfo to a map
+func (c ChangeInfo) ToMap() map[string]interface{} {
+	return map[string]interface{}{
+		"actor":      c.Actor,
+		"actor_type": c.ActorType,
+		"timestamp":  c.Timestamp,
+	}
+}
+
+// PolicySetPolicy represents a policy in the policy set
+type PolicySetPolicy struct {
+	Type string `json:"type,omitempty"`
+	ID   string `json:"id,omitempty"`
+}
+
+// ToMap converts PolicySetPolicy to a map
+func (p PolicySetPolicy) ToMap() map[string]interface{} {
+	return map[string]interface{}{
+		"type": p.Type,
+		"id":   p.ID,
+	}
+}
+
+// Scope represents the scope of the policy set
+type Scope struct {
+	RepoIds []string `json:"repoIds,omitempty"`
+}
+
+// ToMap converts Scope to a list of maps
+func (s *Scope) ToMap() []map[string]interface{} {
```

Comment:
Policy uses list, rego policies set, I think I had a discussion on this with Pedro but can't recall. Right now it's aligned with other resources we have, not sure we want to change this? 
@wcmjunior any input here?

Here is a test I did. 
Both the resource and the dataset use type list (as policy resource/datasource also does).
And it works as expected on create, eg you do:
```
resource "cyral_policy_set" "repo_lockdown_example" {
  wizard_id    = "repo-lockdown"
  name        = "no block"
  description = "This default policy will block by default all queries for myrepo"
  enabled = true
  tags = ["default block", "fail closed"]
    scope {
    repo_ids = [cyral_repository.myrepo.id,cyral_repository.myrepo.id]
  }
  wizard_parameters    = jsonencode({
    denyByDefault = true
    failClosed = true
    })
}

```which will create a set with those two repos in scope. Reading it back gives you this.
```
terraform apply
cyral_repository.myrepo2: Refreshing state... [id=2pRnxNKLVIvEnn2t5ZRRRfTICwO]
cyral_repository.myrepo: Refreshing state... [id=2pRnxKQivus0V1W884De2Bb9shX]
cyral_policy_set.repo_lockdown_example: Refreshing state... [id=eba9157f-ff97-4abc-ae9e-9c48d4eedb03]
data.cyral_policy_set.repo_lockdown_example: Reading...
data.cyral_policy_set.repo_lockdown_example: Read complete after 0s [id=eba9157f-ff97-4abc-ae9e-9c48d4eedb03]

Changes to Outputs:
  + description       = "This default policy will block by default all queries for myrepo"
  + name              = "no block"
  + scope             = [
      + {
          + repo_ids = [
              + "2pRnxKQivus0V1W884De2Bb9shX",
              + "2pRnxKQivus0V1W884De2Bb9shX",
            ]
        },
    ]
  + wizard_parameters = jsonencode(
        {
          + denyByDefault = true
          + failClosed    = true
        }
    )

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

description = "This default policy will block by default all queries for myrepo"
name = "no block"
scope = tolist([
  {
    "repo_ids" = tolist([
      "2pRnxKQivus0V1W884De2Bb9shX",
      "2pRnxKQivus0V1W884De2Bb9shX",
    ])
  },
])
wizard_parameters = "{\"denyByDefault\":true,\"failClosed\":true}"

```

Reply:
The SDK does not support defining a TypeMap with an Elem of *schema.Resource{}. Attempting to do so results in an error:
`TypeMap with Elem *Resource not supported, use TypeList/TypeSet with Elem *Resource or TypeMap with Elem *Schema`
The scope field represents a nested block that can contain multiple attributes and potentially be extended in the future. Using a TypeList with an Elem of *schema.Resource{} allows us to define a complex nested structure and is therefore what we need to do here (or possibly go with set).

Reply:
Would TypeMap with Elem *Schema work? Sorry this may be a dumb question, I don't quite understand the difference between Resource and Schema in this context.